pkgs/games/armagetronad/default.nix

```
@@ -28,64 +28,65 @@ let
   latestVersionMajor = "0.2.9";
   unstableVersionMajor = "0.4";
 
-  latestCommonBuildInputs = [ SDL SDL_image SDL_mixer libpng ];
-
-  unstableCommonBuildInputs = [ SDL2 SDL2_image SDL2_mixer glew ftgl freetype ];
-  unstableCommonNativeBuildInputs = [ SDL ]; # for sdl-config
-
-  srcs = {
-    ${latestVersionMajor} = rec {
-      version = "${latestVersionMajor}.1.1";
-      src = fetchFromGitLab {
+  srcs =
+  let
+    fetchArmagetron = rev: sha256:
+      fetchFromGitLab {
         owner = "armagetronad";
         repo = "armagetronad";
+        inherit rev sha256;
+      };
+  in
+  {
+    # https://gitlab.com/armagetronad/armagetronad/-/tags
+    ${latestVersionMajor} =
+      let
+        version = "${latestVersionMajor}.1.1";
         rev = "v${version}";
         sha256 = "tvmKGqzH8IYTSeahc8XmN3RV+GdE5GsP8pAlwG8Ph3M=";
+      in dedicatedServer: {
+        inherit version;
+        src = fetchArmagetron rev sha256;
+        extraBuildInputs = lib.optionals (!dedicatedServer) [ libpng SDL SDL_image SDL_mixer ];
       };
-      extraBuildInputs = latestCommonBuildInputs;
-    };
 
+    # https://gitlab.com/armagetronad/armagetronad/-/commits/trunk/?ref_type=heads
     ${unstableVersionMajor} =
       let
-        rev = "4bf6245a668ce181cd464b767ce436a6b7bf8506";
-      in
-      {
+        rev = "e7f41fd26363e7c6a72f0c673470ed06ab54ae08";
+        sha256 = "Uxxk6L7WPxKYQ4CNxWwEtvbZjK8BqYNTuwwdleZ44Ro=";
+      in dedicatedServer: {
         version = "${unstableVersionMajor}-${builtins.substring 0 8 rev}";
-        src = fetchFromGitLab {
-          owner = "armagetronad";
-          repo = "armagetronad";
-          inherit rev;
-          sha256 = "cpJmQHCS6asGasD7anEgNukG9hRXpsIJZrCr3Q7uU4I=";
-        };
-        extraBuildInputs = [ protobuf boost ] ++ unstableCommonBuildInputs;
-        extraNativeBuildInputs = [ bison ] ++ unstableCommonNativeBuildInputs;
+        src = fetchArmagetron rev sha256;
+        extraBuildInputs = [ protobuf boost ]
+          ++ lib.optionals (!dedicatedServer) [ glew ftgl freetype SDL2 SDL2_image SDL2_mixer ];
+        extraNativeBuildInputs = [ bison ];
       };
 
+    # https://gitlab.com/armagetronad/armagetronad/-/commits/hack-0.2.8-sty+ct+ap/?ref_type=heads
     "${latestVersionMajor}-sty+ct+ap" =
       let
-        rev = "fdfd5fb97083aed45467385b96d50d87669e4023";
-      in
-      {
+        rev = "a5bffe9dda2b43d330433f76f14eb374701f326a";
+        sha256 = "cNABxfg3MSmbxU/R78QyPOMwXGqJEamaFOPNw5yhDGE=";
```

```suggestion
        hash = "sha256-cNABxfg3MSmbxU/R78QyPOMwXGqJEamaFOPNw5yhDGE=";
```
please don't strip the prefixes of sri hashes